8/SiloHost/src/Program.cs

```diff
@@ -9,70 +8,29 @@
-
-            var siloPort = int.Parse(extractedSiloPort);
-            var developmentPeerPort = int.Parse(extractedPrimaryPort);
-            var gatewayPort = int.Parse(extractedGatewayPort);
-            var dashboardPort = int.Parse(extractDashboardPort);
-            var primaryIp = IPAddress.Parse(primaryAddress);
-
-            var primarySiloEndpoint = new IPEndPoint(primaryIp, developmentPeerPort);
-
-            var siloEndpointConfiguration = new SiloEndpointConfiguration(advertisedIpAddress, siloPort, gatewayPort);
+            ISiloConfigurationHelper siloConfigurationHelper = new SiloConfigurationHelper(environmentVariable);
 
             return new HostBuilder()
                 .UseOrleans(siloBuilder =>
                 {
                     siloBuilder.UseLinuxEnvironmentStatistics();
-                    siloBuilder.UseDashboard(dashboardOptions =>
-                    {
-                        dashboardOptions.Username = "piotr";
-                        dashboardOptions.Password = "orleans";
-                        dashboardOptions.Port = dashboardPort;
-                    });
-
+                    siloBuilder.UseDashboard(dashboardOptions => siloConfigurationHelper.ConfigureDashboardOptions(dashboardOptions));
                     //Register silo with dynamo cluster
-                    siloBuilder.UseDynamoDBClustering(builder =>
-                    {
-                        //Connect to membership table in dynamo
-                        builder.TableName = membershipTable;
-                        builder.Service = awsRegion;
-                    });
-                    siloBuilder.Configure<ClusterOptions>(clusterOptions =>
-                    {
-                        clusterOptions.ClusterId = "cluster-of-silos";
-                        clusterOptions.ServiceId = "hello-world-service";
-                    });
-                    siloBuilder.Configure<EndpointOptions>(endpointOptions =>
-                    {
-                        endpointOptions.AdvertisedIPAddress = siloEndpointConfiguration.Ip;
-                        endpointOptions.SiloPort = siloEndpointConfiguration.SiloPort;
-                        endpointOptions.GatewayPort = siloEndpointConfiguration.GatewayPort;
-                        endpointOptions.SiloListeningEndpoint = new IPEndPoint(IPAddress.Any, siloEndpointConfiguration.SiloPort);
-                        endpointOptions.GatewayListeningEndpoint = new IPEndPoint(IPAddress.Any, siloEndpointConfiguration.GatewayPort);
-                    });
+                    siloBuilder.UseDynamoDBClustering(builder => siloConfigurationHelper.ConfigureDynamoClusterOptions(builder));
+                    siloBuilder.Configure<ClusterOptions>(clusterOptions => siloConfigurationHelper.ConfigureClusterOptions(clusterOptions));
+                    siloBuilder.Configure<EndpointOptions>(options => siloConfigurationHelper.ConfigureEndpointOptions(options));
```

Comment:
Same here - we could have our own extensions for increased readability. This is already good, but we can make it much better.